### PR TITLE
Refactor: Convert CreateUser to use react hook form, MUI, and a semantic html wrapper

### LINKED ISSUE #21: CreateUser <form> integration
In `src/profile/CreateUser` let's add a <form> html component to wrap the form, so that once all the form fields have been validated, we can use the Enter button to submit the new profile information. This is also good for screen-readers + accessibility.

### PR DESCRIPTION
**Pull request by jayeclark**
Pull-Request for `paretOS`

## Description
- Refactors CreateUser to use semantic HTML, react-hook-form, and MUI 
- Fixes some formatting issues 
- Fixes T&C background
- Converts to TypeScript
- Going to open a separate issue for this, but - if you wander away from this page and don't complete the signup process, you'll get logged in and taken to the main screen as Vilfredo (with no user data) rather than being taken to the onboarding page.

## Relates to
Fixes #21 

## Reviewers
- @mikhael28 

## Screenshots / other info
Screen Shot 2022-05-08 at 4 40 45 PM

Screen Shot 2022-05-08 at 4 45 28 PM

### CODE REVIEW

File: src/profile/TermsOfService.js [hunk 1]
```diff
@@ -8,10 +9,17 @@ export default function TermsOfService({
...
+  const theme = useTheme();
   return (
     <>
       <Modal open={open} onClose={onClose}>
-        <>
+        <div
+          style={{
+            margin: -24,
+            padding: 24,
+            backgroundColor: theme.palette.background.paper,
```
Comment (mikhael28) [inline src/profile/TermsOfService.js:20]: ty, this will make the TOS readable :)

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "history": "5.3.0",
     "lodash.clonedeep": "4.5.0",
     "lodash.sortby": "4.7.0",
+    "nanoid": "^3.3.4",
```
Comment (mikhael28) [inline package.json:30]: This is great, thank you for replacing `uuid` :D

File: src/profile/CreateUser.tsx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+import { SyntheticEvent, useState } from "react";
+import {
+  FormControl,
+  TextField,
+  InputLabel,
+  Select,
+  MenuItem,
+  FormHelperText,
+} from "@mui/material";
+import { useForm, Controller } from "react-hook-form";
```
Comment (mikhael28) [inline src/profile/CreateUser.tsx:13]: Great work, thank you! Everything seems to be working great.

**Review by mikhael28 — APPROVED** [review]
Approved! Everything seems to be in order :D

**Comment by mikhael28** [comment]
@jayeclark resolve the merge conflicts as best as possible - it might be worth deleting node_modules and the yarn.lock, and running a clean install with the latest dependencies and then pushing that up for the yarn.lock once you verify everything is in working condition.

**Comment by jayeclark** [comment]
Hmmm @mikhael28 let me figure out why it's failing CI before we merge :)

**Pull request merged**